backend/prompt_studio/prompt_studio/views.py

```diff
@@ -41,3 +47,115 @@ def get_queryset(self) -> Optional[QuerySet]:
+                    "Decrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__gt": start_sequence_number,
+                    "sequence_number__lte": drop_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
+                    filters, increment=False
+                )
+
+            elif start_sequence_number > drop_sequence_number:
+                logger.info(
+                    "Start sequence number is greater than drop sequence number. "
+                    "Incrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__lt": start_sequence_number,
+                    "sequence_number__gte": drop_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
+                    filters, increment=True
+                )
+
+            # If the start and drop sequence numbers are equal,
+            # return a bad request response
+            else:
+                logger.warning(
+                    "Start and drop sequence numbers are equal. "
+                    "Returning bad request response."
+                )
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={"detail": "Start and drop sequence numbers are equal."},
+                )
+
+            # Update the sequence number of the moved prompt
+            prompt_instance.sequence_number = drop_sequence_number
+            prompt_instance.save()
+            # Append the updated prompt instance data to the response
+            filtered_prompts_data.append(
+                {
+                    "id": prompt_instance.prompt_id,
+                    "sequence_number": prompt_instance.sequence_number,
+                }
+            )
+
+            logger.info("Re-ordering completed successfully.")
+            return Response(status=status.HTTP_200_OK, data=filtered_prompts_data)
+
+        except ToolStudioPrompt.DoesNotExist:
+            logger.error(f"Prompt with ID {prompt_id} not found.")
+            return Response(
+                status=status.HTTP_404_NOT_FOUND, data={"detail": "Prompt not found."}
+            )
+
+        except Exception as e:
+            logger.exception(
+                f"An unexpected error occurred while re-ordering the prompts: {e}"
+            )
+            return Response(
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                data={"detail": "An unexpected error occurred."},
+            )
```

Comment:
```suggestion
```
@tahierhussain let's not add such generic exception blocks. This would catch all errors coming from within (for eg, `ValidationError`) and it would suppress that and throw a different error

Reply:
@chandrasekharan-zipstack Thanks for the suggestion. Done.